internal/transactionsubmission/transaction_worker.go

```diff
@@ -184,54 +192,43 @@ func (tw *TransactionWorker) runJob(ctx context.Context, txJob *TxJob) error {
 func (tw *TransactionWorker) handleFailedTransaction(ctx context.Context, txJob *TxJob, hTxResp horizon.Transaction, hErr *utils.HorizonErrorWrapper) error {
 	log.Ctx(ctx).Errorf("🔴 Error processing job: %v", hErr)
 
-	metricsMetadata := tssMonitor.TxMetadata{
-		EventID:          tw.jobUUID,
-		SrcChannelAcc:    txJob.ChannelAccount.PublicKey,
-		PaymentEventType: sdpMonitor.PaymentMarkedForReprocessingLabel,
+	handler, err := tw.handlerFactory.GetHandler(&txJob.Transaction)
+	if err != nil {
+		return fmt.Errorf("getting transaction handler: %w", err)
 	}
-	defer func() {
-		tw.monitorSvc.LogAndMonitorTransaction(
-			ctx,
-			txJob.Transaction,
-			sdpMonitor.PaymentErrorTag,
-			metricsMetadata,
-		)
-	}()
 
-	err := tw.saveResponseXDRIfPresent(ctx, txJob, hTxResp)
+	isRetryable := !(hErr.IsHorizonError() && hErr.ShouldMarkAsError())
+	defer handler.MonitorTransactionProcessingFailed(ctx, txJob, tw.jobUUID, isRetryable, hErr.Error())
+
+	err = tw.saveResponseXDRIfPresent(ctx, txJob, hTxResp)
 	if err != nil {
 		return fmt.Errorf("saving response XDR: %w", err)
 	}
 
 	var hErrWrapper *utils.HorizonErrorWrapper
 	if errors.As(hErr, &hErrWrapper) {
 		if hErrWrapper.IsHorizonError() {
-			metricsMetadata.IsHorizonErr = true
 			tw.txProcessingLimiter.AdjustLimitIfNeeded(hErrWrapper)
 
-			if hErrWrapper.ShouldMarkAsError() {
-				metricsMetadata.PaymentEventType = sdpMonitor.PaymentFailedLabel
-
-				// Building the payment completed event before updating the transaction status. This way, if the message
+			if hErr.IsHorizonError() && hErr.ShouldMarkAsError() {
```

Comment:
You're already under an iff statement where `IsHorizonError()` is true, so you can remove that check here.

```suggestion
			if hErr.ShouldMarkAsError() {
```